### DISCUSSION TITLE
DNS Migration from Netdot

### DISCUSSION THREAD

**Discussion by lairsdragon** [sys4/netbox-plugin-dns#21] [General]
Hi 

I am working on the Migration from Netdot to Netbox. We are using Netdot for IPAM and DNS. IPAM Migration is mostly straight forward exporting from Netdot as CSV and doing some data adaption in this step.

DNS is not as straightforward as I hoped to import just the generated Zone Files to netbox-plugin-dns. Has anybody a hint or sacript for importing a Zone?

**Comment by peteeckel** [comment]
Hi @lairsdragon, the Problem with zone files is in my experience that they are usually hand-edited and the various name servers created a ton of specific syntactic helpers that make them pretty hard to parse and canonicalise automatically. This might not apply to your case, however, as you are using Netdot for populating DNS.

If I had to create a script right now I'd go a slightly different route and make AXFR requests to the DNS server currently hosting the zone and work with the results, which are very clean and usable. The disadvantage is that, for instance, AXFR fills in the TTL for all records, which is probably not what you want, so some tweaking of the data will be necessary.

Can't you export the zone data from Netdot the same way as the IPAM data?

**Reply by lairsdragon** [reply]
I am currently looking into writing my own Exporter for the DNS Data based upon the BIND Exporter of Netdot: https://github.com/cvicente/Netdot/blob/master/lib/Netdot/Exporter/BIND.pm 
That will be no fun but better than massaging 12000 Records by hand..

**Reply by peteeckel** [reply]
Let me know if I can help you when you import the data - depending on how many records you try to import at once, you might run into performance issues and/or timeouts. This is something that has not been thoroughly tested yet, and I'm interested in improving the import mechanism (one vague idea is to optionally offload PTR record generation to `netbox-rq`).

**Comment by kollross** [comment]
I would be interested as well.   I'm doing the same migration (Netdot -> Netbox).   I've found the IPAM CVS export lacking, as it seems to only export the current viewed subnets (doesn't traverse down into containers).

Netdot seems to be only able to export DNS information in BIND format, so trying to figure out how to get that into a format that was feasible to import (either via ansible or something else).   I had looked into OctoDNS to see if I could use that to munge the data into something that Netbox could ingest, but that seemed not as straightforward as I would have hoped.

**Reply by lairsdragon** [reply]
@kollross For the IPAM Data I have build the following quickndirty Hack based upon a perl script from Netdot:

https://gist.github.com/lairsdragon/23d988c7c6115dfaa2a7343df84f5a81

Maybe it is helpful for you..

**Comment by peteeckel** [comment]
OTOH, if Netdot exports as BIND files the resulting BIND files should at least be in a sufficiently sensible and canonicalised format to make parsing and converting them to CSV/JSON/YAML relatively easy, so that might be your best bet.

**Comment by kollross** [comment]
@lairsdragon Did you figure out method for getting DNS information from Netdot to Netbox.

The part I'm struggling to figure out a good solution is splitting up records based on if its a record that needs to go into netbox-dns (example: MX, CNAME, special case A records) vs updating the DNS Name: field in the IPAM  IP address section, then having scripts populate the netbox-dns plugin.
